lib/kafka/connection.rb

```diff
@@ -99,6 +99,34 @@ def send_request(request)
       raise ConnectionError, "Connection error: #{e}"
     end
 
+    def send_async_request(request)
+      # Default notification payload.
+      notification = {
+        broker_host: @host,
+        api: Protocol.api_name(request.api_key),
+        request_size: 0,
+        response_size: 0,
+      }
+
+      @instrumenter.instrument("async_request.connection", notification) do
+        open unless open?
+
+        @correlation_id += 1
```

Comment:
There's an obvious race condition here, of course. We'd need to register the response futures on the connection and deliver the actual responses to them if the blocking reads happen in the wrong order...